src/main/java/edu/usf/cutr/gtfsrtvalidator/api/resource/GtfsRtFeed.java

```diff
@@ -132,56 +131,79 @@ public Response getRtFeeds() {
     private static HashMap<String, ScheduledExecutorService> runningTasks = new HashMap<>();
 
     @PUT
-    @Path("/{id}/{updateInterval}/monitor")
-    public Response getID(@PathParam("id") int id, @PathParam("updateInterval") int updateInterval) {
+    @Path("/{id}/{clientId}/{updateInterval}/monitor")
+    public Response getID(
+            @PathParam("id") int id,
+            @PathParam("clientId") String clientId,
+            @PathParam("updateInterval") int updateInterval) {
         // Store the timestamp when we start monitoring feeds that can be used to query database
         currentTimestamp = System.currentTimeMillis();
         //Get RtFeedModel from id
         Session session = GTFSDB.initSessionBeginTrans();
         GtfsRtFeedModel gtfsRtFeed = (GtfsRtFeedModel) session.createQuery(" FROM GtfsRtFeedModel "
                 + "WHERE rtFeedID = "+id).uniqueResult();
+
+        // Save the session data of a client monitoring feeds.
+        SessionModel sessionModel = new SessionModel();
+        sessionModel.setUserId(clientId);
+        sessionModel.setSessionStartTime(currentTimestamp);
+        sessionModel.setGtfsRtFeedModel(gtfsRtFeed);
+
+        session.save(sessionModel);
         GTFSDB.commitAndCloseSession(session);
 
         //Extract the Url and gtfsId to start the background process
         startBackgroundTask(gtfsRtFeed, updateInterval);
 
-        return Response.ok(gtfsRtFeed, MediaType.APPLICATION_JSON).build();
+        return Response.ok(sessionModel, MediaType.APPLICATION_JSON).build();
     }
 
     // Get Monitor data for requested gtfsRtId
     @GET
     @Path("/{id : \\d+}/summary/pagination/{summaryCurPage: \\d+}/{summaryRowsPerPage: \\d+}" +
-            "/log/{toggledData: .*}/pagination/{logCurPage: \\d+}/{logRowsPerPage: \\d+}")
+            "/log/{toggledData: .*}/pagination/{logCurPage: \\d+}/{logRowsPerPage: \\d+}" +
+            "/{startTime}/{endTime}")
     @Produces(MediaType.APPLICATION_JSON)
     public Response getMonitorData(
             @PathParam("id") int gtfsRtId,
             @PathParam("summaryCurPage") int summaryCurPage,
             @PathParam("summaryRowsPerPage") int summaryRowsPerPage,
             @PathParam("toggledData") String hideErrors,
             @PathParam("logCurPage") int logCurPage,
-            @PathParam("logRowsPerPage") int logRowsPerPage) {
+            @PathParam("logRowsPerPage") int logRowsPerPage,
```

Comment:
Seems like ID should be a path parameter here and everything else a query parameter.